miner/Miner.hs

```diff
@@ -263,7 +285,72 @@ scheme :: Env -> (TargetBytes -> HeaderBytes -> RIO Env HeaderBytes)
 scheme env = case envCmd env of
     CPU e _ -> cpu e
     GPU e _ -> gpu e
-    Keys -> error "Impossible: You shouldn't reach this case."
+    _ -> error "Impossible: You shouldn't reach this case."
+
+
+{- this will be ugly -}
+genLocal
+    :: ChainwebVersion
+    -> ChainId
+    -> P.Command Text
+    -> ClientM (P.CommandResult Hash)
+genLocal version cid =
+  case someChainwebVersionVal version of
+    SomeChainwebVersionT (_ :: Proxy cv) ->
+      case someChainIdVal cid of
+        SomeChainIdT (_ :: Proxy ccid) ->
+          client (Proxy :: Proxy (PactLocalApi cv ccid))
```

Comment:
This code exists. See https://github.com/kadena-io/chainweb-node/blob/master/src/Chainweb/Pact/RestAPI/Client.hs#L79